src/export/dashboard.ts

```diff
@@ -0,0 +1,72 @@
+// (C) 2019 GoodData Corporation
+import { DashboardExport } from "@gooddata/typings";
+import { ApiResponse, XhrModule } from "../xhr";
+import { IExportResponse } from "../interfaces";
+import { handleHeadPolling, IPollingOptions } from "../util";
+import { isExportFinished, sanitizeDateFilters } from "./util";
+
+interface IDashboardExportPayload {
+    dashboardExport: {
+        dashboardUri: string;
+        filters?: DashboardExport.FilterContextItem[];
+    };
+}
+
+export class DashboardExportModule {
+    constructor(private xhr: XhrModule) {}
+
+    public async toPdf(
+        projectId: string,
+        dashboardUri: string,
+        filters: DashboardExport.FilterContextItem[] = [],
+        pollingOptions: IPollingOptions = {},
+    ): Promise<IExportResponse> {
+        const sanitizedFilters = sanitizeDateFilters(filters);
+        const payload = this.getDashboardExportPayload(dashboardUri, sanitizedFilters);
+
+        try {
+            const response: ApiResponse = await this.xhr.post(
+                `/gdc/internal/projects/${projectId}/exportDashboard`,
+                { body: payload },
+            );
+            return await this.pollPdfFile(response, pollingOptions);
```

Comment:
I think this should be without `await` as return type is still expecting Promise